pkg/resources/view.go

```diff
@@ -237,63 +622,130 @@ func UpdateView(d *schema.ResourceData, meta interface{}) error {
 		if comment := d.Get("comment").(string); comment == "" {
 			err := client.Views.Alter(ctx, sdk.NewAlterViewRequest(id).WithUnsetComment(true))
 			if err != nil {
-				return fmt.Errorf("error unsetting comment for view %v", d.Id())
+				return diag.FromErr(fmt.Errorf("error unsetting comment for view %v", d.Id()))
 			}
 		} else {
 			err := client.Views.Alter(ctx, sdk.NewAlterViewRequest(id).WithSetComment(comment))
 			if err != nil {
-				return fmt.Errorf("error updating comment for view %v", d.Id())
+				return diag.FromErr(fmt.Errorf("error setting comment for view %v", d.Id()))
 			}
 		}
 	}
 
 	if d.HasChange("is_secure") {
-		if d.Get("is_secure").(bool) {
-			err := client.Views.Alter(ctx, sdk.NewAlterViewRequest(id).WithSetSecure(true))
+		if v := d.Get("is_secure").(string); v != BooleanDefault {
+			parsed, err := booleanStringToBool(v)
 			if err != nil {
-				return fmt.Errorf("error setting secure for view %v", d.Id())
+				return diag.FromErr(err)
+			}
+			if parsed {
+				err = client.Views.Alter(ctx, sdk.NewAlterViewRequest(id).WithSetSecure(parsed))
+			} else {
+				err = client.Views.Alter(ctx, sdk.NewAlterViewRequest(id).WithUnsetSecure(parsed))
+			}
+			if err != nil {
+				return diag.FromErr(fmt.Errorf("error setting is_secure for view %v: %w", d.Id(), err))
 			}
 		} else {
 			err := client.Views.Alter(ctx, sdk.NewAlterViewRequest(id).WithUnsetSecure(true))
 			if err != nil {
-				return fmt.Errorf("error unsetting secure for view %v", d.Id())
+				return diag.FromErr(fmt.Errorf("error unsetting is_secure for view %v: %w", d.Id(), err))
 			}
 		}
 	}
-
-	if d.HasChange("tag") {
-		unsetTags, setTags := GetTagsDiff(d, "tag")
-
-		if len(unsetTags) > 0 {
-			err := client.Views.Alter(ctx, sdk.NewAlterViewRequest(id).WithUnsetTags(unsetTags))
+	if d.HasChange("change_tracking") {
+		if v := d.Get("change_tracking").(string); v != BooleanDefault {
+			parsed, err := booleanStringToBool(v)
+			if err != nil {
+				return diag.FromErr(err)
+			}
+			err = client.Views.Alter(ctx, sdk.NewAlterViewRequest(id).WithSetChangeTracking(parsed))
 			if err != nil {
-				return fmt.Errorf("error unsetting tags on %v, err = %w", d.Id(), err)
+				return diag.FromErr(fmt.Errorf("error setting change_tracking for view %v: %w", d.Id(), err))
+			}
+		} else {
+			err := client.Views.Alter(ctx, sdk.NewAlterViewRequest(id).WithSetChangeTracking(false))
+			if err != nil {
+				return diag.FromErr(fmt.Errorf("error unsetting change_tracking for view %v: %w", d.Id(), err))
 			}
 		}
+	}
 
-		if len(setTags) > 0 {
-			err := client.Views.Alter(ctx, sdk.NewAlterViewRequest(id).WithSetTags(setTags))
+	if d.HasChange("row_access_policy") {
+		var addReq *sdk.ViewAddRowAccessPolicyRequest
+		var dropReq *sdk.ViewDropRowAccessPolicyRequest
+		if v, ok := d.GetOk("row_access_policy"); ok {
+			rowAccessPolicyConfig := v.([]any)[0].(map[string]any)
```

Comment:
I'm wondering if `ListDiff` function that we have in helpers could simplify this a bit, wdyt?

Reply:
Not really, because these are pseudo-lists, and can have max 1 element. But I simplified the code though.